network/src/primary.rs

```diff
@@ -263,3 +107,119 @@ impl UnreliableNetwork for PrimaryToWorkerNetwork {
         handler
     }
 }
+
+pub struct PrimaryNetwork {
+    network: anemo::Network,
+    retry_config: RetryConfig,
+    /// Small RNG just used to shuffle nodes and randomize connections (not crypto related).
+    rng: SmallRng,
+    // One bounded executor per address
+    executors: HashMap<anemo::PeerId, BoundedExecutor>,
+}
+
+impl PrimaryNetwork {
+    pub fn new(network: anemo::Network) -> Self {
+        let retry_config = RetryConfig {
+            // Retry for forever
+            retrying_max_elapsed_time: None,
+            ..Default::default()
+        };
+
+        Self {
+            network,
+            retry_config,
+            rng: SmallRng::from_entropy(),
+            executors: HashMap::new(),
+        }
+    }
+
+    pub fn cleanup<'a, I>(&mut self, _to_remove: I)
+    where
+        I: IntoIterator<Item = &'a Multiaddr>,
+    {
+        // do nothing atm
```

Comment:
Confused what this is? Should it maybe be turned into a TODO or removed?

Reply:
Yeah i'll add a TODO